src/ir/attribute.rs

```diff
@@ -0,0 +1,94 @@
+use crate::prelude::*;
+
+use crate::ir::Identifier;
+use crate::ir::Literal;
+
+use syn::{AttributeArgs, Meta, MetaList, MetaNameValue, NestedMeta, Path};
+
+/// Attribute Enum
+#[derive(Debug)]
+pub enum Attribute {
+    /// Literal Variant
+    Literal(Literal),
+    /// Named Variant
+    Named(Identifier, Literal),
+    /// Group Variant
+    Group(Identifier, Attributes),
+}
+
+#[derive(Shrinkwrap, Default, Debug)]
+#[shrinkwrap(mutable)]
+/// Attributes Struct
+pub struct Attributes {
+    /// attributes field
+    pub attributes: Vec<Attribute>,
+}
+
+impl From<AttributeArgs> for Attributes {
+    fn from(attribute_args: AttributeArgs) -> Self {
+        println!("attribute_args: {:#?}", attribute_args);
+        let mut vec: Vec<Attribute> = vec![];
+        for arg in attribute_args {
+            vec.push(Attribute::from(arg));
+        }
+        Self { attributes: vec }
```

Comment:
This can be rewritten as
```rust
let attributes: Vec<_> = attribute_args.iter().map(Attribute::from).collect();
Self { attributes }
```

Reply:
Changed in [commit b671c47](https://github.com/sensorial-systems/ligen-core/pull/21/commits/b671c47ade5d37205a4073ebe214074048ecbc35)